packages/opentelemetry-node/src/instrumentation/PluginLoader.ts

```diff
@@ -78,6 +78,16 @@ export class PluginLoader {
         return this;
       }
 
+      const alreadyRequiredModules = Object.keys(require.cache);
```

Comment:
IMO this logic deserved to be in a separate function (`logModulesLoadedBeforeSDKsetup`) WDYT?

Reply:
Should this be a `Set` instead of an array?

Reply:
@dyladan Any module can only have one instance in the cache, so in theory it's not possible to have duplicates. Otherwise i don't see the point of using a `Set` there, can you elaborate ?

@mayurkale22 IMO The logic is pretty simple i don't think it's clearer to split this function

Reply:
@vmarchaud Just that a Set has `O(1)` existence checks. Not sure how checking for the existence of a string in an array affects startup for applications with large dependency trees. Hopefully we should be one of the first things loaded so the list of already loaded dependencies should be small, but I don't see a reason to _not_ use a Set.

Reply:
Actually it's not possible, the path in `require.cache` are absolute (and it contains the javascript file) so i must iterate on each one to see if it contains the name of the module we are trying to hook.

Reply:
ah I understand